File: typescript/armstrong-numbers/armstrong-numbers.ts
function isArmstrongNumber(n: number): boolean {
  const len = Math.ceil( Math.log10( n ))
  let armstrongSum = 0
  let tmp = n
  while (tmp > 0) {
    const digit = tmp % 10
    armstrongSum += digit ** len
    tmp = (tmp - digit) / 10
  }
  return armstrongSum === n
}

export default { isArmstrongNumber }

File: python/armstrong-numbers/armstrong_numbers.py
from functools import reduce
import math


def is_armstrong_number(number):
    if number <  0: raise ValueError('Non-negative numbers only')
    if number == 0: return True

    n = math.ceil(math.log10(number))
    sum = reduce(lambda s, d: s + d**n, map(int, str(number)), 0)

    '''
    # iterative
    tmp = number
    sum = 0
    while tmp > 0:
        tmp, digit = divmod(tmp, 10)
        sum += digit ** n
    '''

    return number == sum
